feat: 🍰 Post Editor Legend

Blargian opened the pull request:
## 🍰 Pullrequest

Work in progress

Added a new component called MenuLegend (adapted the content menu component) with localised content for Deutsch, English and Russian. The legend displays when the user hovers over it (desktop) or when the user clicks on the button (mobile/tablet). 

![image](https://user-images.githubusercontent.com/41984034/123546743-315ce000-d75e-11eb-98b8-ddec34511f34.png)

### Issues
<!-- Which Issues does this fix, which are related?
- fixes #3673
- relates #3673 
-->

fixes #3673

### Todo
<!-- In case some parts are still missing, list them here. -->
- [X] Add a component for the menu legend
- [X] Add a question-mark icon on the editor menu
- [x] Display it on hover rather than on click
- [x] Make it multilingual
- [x] Make it responsive
- [x] Fix failing tests

Tirokk commented on package.json:
```diff
@@ -49,5 +49,8 @@
   },
   "resolutions": {
     "set-value": "^2.0.1"
+  },
+  "dependencies": {
+    "jq": "^1.7.2"
```
I don’t understand where all the dependencies are coming from, because you didn’t invent anything new, as far I can see …
Here and the others below … 👇🏼

Any ideas? @Blargian @Mogge

Blargian replied:
@Tirokk this one @Mogge said I would need when he helped me get the dev environment setup. It's required for implementing multilanguage support or for testing related to that. 😃

Tirokk replied:
Hey @Mogge , isn’t this needed for the webapp only? And not for the main folder … 🤔

Mogge replied:
It runs over the `scripts´ folder in the main folder. So I guess it is right there. We have to see what happens when running `yarn locales`.

Tirokk commented on webapp/components/Editor/MenuBar.vue:
```diff
@@ -59,22 +59,41 @@
+import MenuLegend from './MenuLegend.vue'
 
 export default {
+  data() {
+    return {}
+  },
```
In case it is empty you can leave it away, I guess. @Blargian 👍🏼

Tirokk commented on yarn.lock:
```diff
@@ -1717,6 +1717,13 @@ binary-extensions@^1.0.0:
   resolved "https://registry.yarnpkg.com/binary-extensions/-/binary-extensions-1.13.0.tgz#9523e001306a32444b907423f1de2164222f6ab1"
   integrity sha512-EgmjVLMn22z7eGGv3kcnHwSnJXmFHjISTY9E/S5lIcTD3Oxw05QTcBLNkJFzcb3cNueUdF/IN4U+d78V0zO8Hw==
 
+bindings@^1.2.1:
+  version "1.5.0"
+  resolved "https://registry.yarnpkg.com/bindings/-/bindings-1.5.0.tgz#10353c9e945334bc0511a6d90b38fbc7c9c504df"
+  integrity sha512-p2q/t/mhvuOj/UeLlV6566GD/guowlr0hHxClI0W9m7MWYkL1F0hLo+0Aexs9HSPCtR1SXQ0TD3MMKrXZajbiQ==
+  dependencies:
+    file-uri-to-path "1.0.0"
+
```
same same

Tirokk commented:
Hey @Blargian !

This looks quite nice for a start. 🤗😍
And for a VueJS beginner … 👍🏼
So I’m looking forward to see your development.

I have left you little hints …

Mogge commented on webapp/components/Editor/MenuLegend.vue:
```diff
@@ -0,0 +1,87 @@
+    return {
+      hover: false,
+      legenditems: [
+        { iconname: 'bold', name: 'bold', shortcut: 'Ctrl+b' },
```
fett

Mogge commented on webapp/components/Editor/MenuLegend.vue:
```diff
@@ -0,0 +1,87 @@
+      hover: false,
+      legenditems: [
+        { iconname: 'bold', name: 'bold', shortcut: 'Ctrl+b' },
+        { iconname: 'italic', name: 'italic', shortcut: 'Ctrl+i' },
```
kursiv

Mogge commented on webapp/components/Editor/MenuLegend.vue:
```diff
@@ -0,0 +1,87 @@
+        { iconname: 'bold', name: 'bold', shortcut: 'Ctrl+b' },
+        { iconname: 'italic', name: 'italic', shortcut: 'Ctrl+i' },
+        { iconname: 'underline', name: 'underline', shortcut: 'Ctrl+u' },
+        { iconname: 'link', name: 'link', shortcut: '' },
```
Unterstrichen

Mogge commented on webapp/components/Editor/MenuLegend.vue:
```diff
@@ -0,0 +1,87 @@
+        { iconname: 'italic', name: 'italic', shortcut: 'Ctrl+i' },
+        { iconname: 'underline', name: 'underline', shortcut: 'Ctrl+u' },
+        { iconname: 'link', name: 'link', shortcut: '' },
+        { iconname: 'paragraph', name: 'paragraph', shortcut: '' },
```
Absatz

Mogge commented on webapp/components/Editor/MenuLegend.vue:
```diff
@@ -0,0 +1,87 @@
+        { iconname: 'underline', name: 'underline', shortcut: 'Ctrl+u' },
+        { iconname: 'link', name: 'link', shortcut: '' },
+        { iconname: 'paragraph', name: 'paragraph', shortcut: '' },
+        { label: 'H3', name: 'heading 3', shortcut: '### + space' },
```
Überschrift 3

Mogge commented on webapp/components/Editor/MenuLegend.vue:
```diff
@@ -0,0 +1,87 @@
+        { iconname: 'link', name: 'link', shortcut: '' },
+        { iconname: 'paragraph', name: 'paragraph', shortcut: '' },
+        { label: 'H3', name: 'heading 3', shortcut: '### + space' },
+        { label: 'H4', name: 'heading 4', shortcut: '#### + space' },
```
Überschrift 4

Mogge commented on webapp/components/Editor/MenuLegend.vue:
```diff
@@ -0,0 +1,87 @@
+        { iconname: 'paragraph', name: 'paragraph', shortcut: '' },
+        { label: 'H3', name: 'heading 3', shortcut: '### + space' },
+        { label: 'H4', name: 'heading 4', shortcut: '#### + space' },
+        { iconname: 'list-ul', name: 'unordered list', shortcut: '* + space' },
```
Ungeordnete Liste

Mogge commented on webapp/components/Editor/MenuLegend.vue:
```diff
@@ -0,0 +1,87 @@
+        { label: 'H3', name: 'heading 3', shortcut: '### + space' },
+        { label: 'H4', name: 'heading 4', shortcut: '#### + space' },
+        { iconname: 'list-ul', name: 'unordered list', shortcut: '* + space' },
+        { iconname: 'list-ol', name: 'ordered list', shortcut: '1. + space' },
```
Geordnete Liste

Mogge commented on webapp/components/Editor/MenuLegend.vue:
```diff
@@ -0,0 +1,87 @@
+        { label: 'H4', name: 'heading 4', shortcut: '#### + space' },
+        { iconname: 'list-ul', name: 'unordered list', shortcut: '* + space' },
+        { iconname: 'list-ol', name: 'ordered list', shortcut: '1. + space' },
+        { iconname: 'quote-right', name: 'quote', shortcut: '> + space' },
```
Zitat

Blargian replied:
> I gave you the German translations for the legend. What happens, if you run `yarn locales --fix` in the webapp folder?

Thanks for the German translations! `yarn locales --fix` runs two scripts in scripts/translations, they both result in "jq: command not found". I tried removing jq from the main package.json file and adding the dependency just to the webapp but it gives the same message.

ulfgebhardt replied:
Could this be related? https://github.com/Ocelot-Social-Community/Ocelot-Social/blob/master/.github/workflows/test.yml#L25
I am not sure about all those translation tools & checks, but we have two scripts executed in our test workflow.
@Mogge

Mogge commented on webapp/components/Editor/MenuLegend.vue:
```diff
@@ -0,0 +1,87 @@
+        { iconname: 'list-ul', name: 'unordered list', shortcut: '* + space' },
+        { iconname: 'list-ol', name: 'ordered list', shortcut: '1. + space' },
+        { iconname: 'quote-right', name: 'quote', shortcut: '> + space' },
+        { iconname: 'minus', name: 'ruler', shortcut: '---' },
```
Linie

Mogge reviewed (commented):
I gave you the German translations for the legend. What happens, if you run `yarn locales --fix` in the webapp folder?

Blargian commented:
I now made the changes so that the legend displays when the user hovers over it. It is also localised for English, Deutsch and Russian. I'm getting some tests failing however - one of the backend unit tests and one of the cyprus tests.

Blargian commented:
Hi @Tirokk, I fixed up one issue which was causing a cypress test to fail. Also removed jq as a dependency all together as it also seemed to be causing some tests to fail.

Tirokk reviewed (approved):
Hey @Blargian ,

I’m back from holidays 🏝 and have reviewed your PR.
Thanks for your patience and sorry for the delay! 🙏🏼

You did a really great job. Very cool 😍🚀💫💫

I have taken the liberty of changing a few small things that I noticed. Nothing serious. 👍🏼
Hope that’s okay for you. Please let me know if you like to care for those things for your own in the future.

Please merge, if the tests are successful! 🥰

Tirokk commented:
Hey dear @Blargian ,
then I merge your PR into our code base. 😍

Blargian commented:
Hi @Tirokk! Sorry for not replying. I saw your message and then got a bit busy and forgot to respond. Of course I don't mind for you to make some changes. 🤗 Thank you

Tirokk commented:
There is an console warning I and @Blargian have overseen: https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/3673#issuecomment-883266073
